nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/media/ExtractMediaMetadata.java

```diff
@@ -0,0 +1,276 @@
+
+        final ProcessorLog logger = this.getLogger();
+        final ObjectHolder<Map<String, String>> value = new ObjectHolder<>(null);
+        final Integer max = context.getProperty(MAX_NUMBER_OF_ATTRIBUTES).asInteger();
+        final String prefix = context.getProperty(METADATA_KEY_PREFIX).evaluateAttributeExpressions().getValue();
+        final FlowFile ff = flowfile;
+
+        try {
+            session.read(flowfile, new InputStreamCallback() {
+                @Override
+                public void process(InputStream in) throws IOException {
+                    try {
+                        Map<String, String> results = tika_parse(ff, in, prefix, max);
+                        value.set(results);
+                    } catch (SAXException | TikaException e) {
+                        throw new IOException(e);
+                    }
+                }
+            });
+
+            // Write the results to attributes
+            Map<String, String> results = value.get();
+            if (results != null && !results.isEmpty()) {
+                flowfile = session.putAllAttributes(flowfile, results);
+            }
+
+            session.transfer(flowfile, SUCCESS);
+        } catch (ProcessException e) {
+            logger.error("Failed to extract media metadata from {} due to {}", new Object[]{flowfile, e});
+            session.transfer(flowfile, FAILURE);
+        }
+    }
+
+    private Map<String, String> tika_parse(FlowFile ff, InputStream sourceStream, String prefix, Integer max) throws IOException, TikaException, SAXException {
+        final Metadata metadata = new Metadata();
+        final TikaInputStream tikaInputStream = TikaInputStream.get(sourceStream);
+        new AutoDetectParser().parse(tikaInputStream, new BodyContentHandler(), metadata);
+        final String content_type = metadata.get(Metadata.CONTENT_TYPE);
+
+        // if parsed MIME type doesn't match filter fail fast without processing attributes
+        final Pattern mimeTypeFilter = mimeTypeFilterRef.get();
+        if (mimeTypeFilter != null && (content_type == null || !mimeTypeFilter.matcher(metadata.get(Metadata.CONTENT_TYPE)).matches())) {
+            return null;
+        }
+
+        final Map<String, String> results = new HashMap<>();
+        final Pattern metadataKeyFilter = metadataKeyFilterRef.get();
+        final StringBuilder dataBuilder = new StringBuilder();
+        final String safePrefix = (prefix == null) ? "" : prefix;
+        for (final String key : metadata.names()) {
+            if (metadataKeyFilter != null && !metadataKeyFilter.matcher(key).matches()) {
+                continue;
+            }
+            dataBuilder.setLength(0);
+            if (metadata.isMultiValued(key)) {
+                for (String val : metadata.getValues(key)) {
+                    if (dataBuilder.length() > 1) {
+                        dataBuilder.append(", ");
+                    }
+                    dataBuilder.append(val);
```

Comment:
How large can multi-valued keys be? Should there be a property for max length?

I am worried about concatenating many values together and creating a huge attribute.